src/lua/zencode_pvss.lua

```diff
@@ -0,0 +1,227 @@
+--GNU Affero General Public License v3.0
+--If not, see http://www.gnu.org/licenses/agpl.txt
+--
+--]]
+
+local PVSS = require'crypto_pvss'
+
+local GENERATORS = PVSS.set_generators()
+
+local function pvss_public_key_f(obj)
+    local point = obj:zcash_topoint()
+    ZEN.assert(
+       point ~=  ECP.infinity(),
+       'pvss public key is not valid (is infinity)'
+    )
+    ZEN.assert(
+        point*ECP.order() == ECP.infinity(),
+        'pvss public key is not valid (not in the subgroup)'
+    )
+    return point
+end
+
+local function import_public_shares_f(obj)
+    local res = {}
+    res.public_keys = ZEN.get(obj, 'public_keys', pvss_public_key_f)
+    res.proof = ZEN.get(obj, 'proof', BIG.new, O.from_base64)
+    res.commitments = ZEN.get(obj, 'commitments', O.zcash_topoint, O.from_base64)
+    res.encrypted_shares = ZEN.get(obj, 'encrypted_shares', O.zcash_topoint, O.from_base64)
+    return res
+end
+
+local function export_public_shares_f(obj)
+    local res = {["public_keys"] = {}, ["proof"] = {}, ["encrypted_shares"] = {}, ["commitments"] = {}}
+    for i = 1, #obj.commitments do
+        res.commitments[i] = obj.commitments[i]:zcash_export()
+    end
+    local n = #obj.public_keys
+    for i=1,n do
+        res.public_keys[i] = obj.public_keys[i]:zcash_export()
+        res.proof[i] = obj.proof[i]:octet():base64()
+        res.encrypted_shares[i] = obj.encrypted_shares[i]:zcash_export()
+    end
+    res.proof[n+1] = obj.proof[n+1]:octet():base64()
+    return res
+end
+
+local function import_secret_share_f(obj)
+    local res = {}
+    res.index = ZEN.get(obj, 'index', INT.from_decimal, tostring)
+    res.proof = ZEN.get(obj, 'proof', BIG.new, O.from_base64)
+    res.dec_share = ZEN.get(obj, 'dec_share', O.zcash_topoint, O.from_base64)
+    res.enc_share = ZEN.get(obj, 'enc_share', O.zcash_topoint, O.from_base64)
+    res.pub_key = ZEN.get(obj, 'pub_key', pvss_public_key_f)
+    return res
+end
+
+local function export_secret_share_f(obj)
+    local res = {["proof"]={}}
+    res.index = obj.index:decimal()
+    res.enc_share = obj.enc_share:zcash_export()
+    res.dec_share = obj.dec_share:zcash_export()
+    res.pub_key = obj.pub_key:zcash_export()
+    for i=1,#obj.proof do
+        res.proof[i] = obj.proof[i]:octet():base64()
+    end
+    return res
+end
+
+local function import_verified_shares_f(obj)
+    local res = {}
+    res.valid_indexes = ZEN.get(obj, 'valid_indexes', INT.from_decimal, tostring)
+    res.valid_shares = ZEN.get(obj, 'valid_shares', O.zcash_topoint, O.from_base64)
+    return res
+end
+
+local function export_verified_shares_f(obj)
+    local res = {["valid_indexes"] = {}, ["valid_shares"] = {}}
+    for i = 1, #obj.valid_indexes do
+        res.valid_shares[i] = obj.valid_shares[i]:zcash_export()
+        res.valid_indexes[i] = obj.valid_indexes[i]:decimal()
+    end
+    return res
+end
+
+local function check_inputs(pks, commitments, encrypted_shares, proof, t, n)
```

Comment:
No need for a separate local function, since this is used only once in the scope of this file, the code can be included directly in the verify statement.